### PR TITLE
Feature/add input variant type underlined

### PR DESCRIPTION
**Pull request by ustcfury**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [✅ ] 🆕 New feature
- [ ] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues
related to https://github.com/ant-design/ant-design/issues/51379 #51379 

### 💡 Background and Solution
问题描述：当前 variant 仅支持 ["outlined", "borderless", "filled"]，可以再增加一个 underlined 的选项让整个表单的输入控件呈现下划线式的风格。
解决方案：在 input/style 文件夹下 variants 中做出改动，新增 underlined 样式，适用于 Input、InputNumber、Mentions 组件。
// todo: Select Cascader TreeSelect 中的 variant 也需要增加 underlined，后续会做出更改。

Background: Currently the variant only supports [‘outlined’, ‘borderless’, ‘filled’], you can add an underlined option to make the whole form's input controls look underlined.
Solution: Make a change in the variants in the input/style folder to add an underlined style for the input, inputNumber, and mentions components.
// todo: The variant in the Select Cascader TreeSelect also needs to be underlined, which will be changed later.

### 📝 Change Log
为 Input、InputNumber、Mentions、DatePicker 组件的 variant 添加 underlined 属性。

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Adds the underlined attribute to the variant of the Input, InputNumber, and Mentions components. |
| 🇨🇳 Chinese | 为 Input、InputNumber、Mentions 组件的 variant 添加 underlined 属性。 |

### CODE REVIEW

File: components/input/style/variants.ts [hunk 1]
```diff
@@ -357,3 +357,107 @@ export const genFilledGroupStyle = (token: InputToken): CSSObject => ({
...
+
+  '&:hover': {
+    borderColor: options.hoverBorderColor,
+    backgroundColor: token.hoverBg,
+  },
+
+  '&:focus, &:focus-within': {
+    borderColor: options.activeBorderColor,
+    // 阴影可以去除
+    // boxShadow: '0 2px 0 rgba(5, 145, 255, 0.1)',
```
[the remark this answers is not in the document]
Comment (ustcfury) [reply components/input/style/variants.ts]: fix it

**Comment by ustcfury** [comment]
有大佬有空帮忙看看嘛 求助！！！！！

File: components/input/style/variants.ts [hunk 2]
```diff
@@ -357,3 +357,100 @@ export const genFilledGroupStyle = (token: InputToken): CSSObject => ({
...
+    hoverBorderColor: string;
+    activeBorderColor: string;
+    activeShadow: string;
+  },
+): CSSObject => ({
+  background: token.colorBgContainer,
+  borderWidth: `0 0 ${token.lineWidth} 0`,
+  borderStyle: `none none ${token.lineType} none`,
+  borderColor: options.borderColor,
+  borderRadius: '0 !important',
```
Comment (MadCcc) [inline components/input/style/variants.ts:376]: 尽量不使用 important

Comment (ustcfury) [reply components/input/style/variants.ts:376]: Input 组件有一个样式 ant-input-affix-wrapper，会覆盖设置 borderRadius 属性，样式顺序那里不敢改怕出问题。大佬还有什么别的方法吗求助

Comment (ustcfury) [reply components/input/style/variants.ts:376]: new https://github.com/ant-design/ant-design/pull/51417

**Comment by ustcfury** [comment]
new https://github.com/ant-design/ant-design/pull/51417

**Pull request closed without merging**